common/s3util.cpp

```diff
@@ -312,6 +314,22 @@ GetObjectMetadataResponse S3Util::getObjectMetadata(const string &key) {
   return GetObjectMetadataResponse(std::move(metadata), "");
 }
 
+GetObjectSizeAndModTimeResponse S3Util::getObjectSizeAndModTime(const string& key) {
+  HeadObjectRequest headObjectRequest;
+  headObjectRequest.SetBucket(bucket_);
+  headObjectRequest.SetKey(key);
+  map<string, uint64_t> metadata;
+  auto headObjectOutcome = s3Client->HeadObject(headObjectRequest);
```

Comment:
how is HeadObject() different from the MakeHttpRequest() call from line 297? Can we reuse getObjectMetadata()?

Reply:
some infomation like the ModTime was not exposed in the MakeHttpRequest() API

Reply:
Is "LastModified" on https://docs.aws.amazon.com/cli/latest/reference/s3api/head-object.html what we need?

Reply:
the format of LastModified by MakeHttpRequest is not easy to convert to the uint64_t format